### PR TITLE
Permit repeating WaitHandle of infinite timeout

### PR DESCRIPTION
**Pull request by AaronRobinsonMSFT**
This issue was found while helping on https://github.com/dotnet/runtime/pull/72051

/cc @roji

### CODE REVIEW

File: src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.WaitThread.cs [hunk 1]
```diff
@@ -373,7 +373,10 @@ private void QueueWaitCompletion(RegisteredWaitHandle registeredHandle, bool tim
                 // If the handle is a repeating handle, set up the next call. Otherwise, remove it from the wait thread.
                 if (registeredHandle.Repeating)
                 {
-                    registeredHandle.RestartTimeout();
+                    if (!registeredHandle.IsInfiniteTimeout)
+                    {
+                        registeredHandle.RestartTimeout();
+                    }
```
Comment (stephentoub) [inline src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.WaitThread.cs:379]: Can we add a test?

Comment (AaronRobinsonMSFT) [reply src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPool.WaitThread.cs:379]: Yes, I will be doing that.

**Review by jkoritzinsky — APPROVED** [review]

**Review by kouvel — APPROVED** [review]
Thanks!

**Comment by AaronRobinsonMSFT** [comment]
Failures are https://github.com/dotnet/runtime/issues/70969

**Review by stephentoub — APPROVED** [review]

**Pull request merged**